bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#include "bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.hpp"
+
+#include "bark/world/observed_world.hpp"
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+EgoRightmostLaneLabelFunction::EgoRightmostLaneLabelFunction(
+    const std::string& label_str, const double distance_thres)
+    : BaseLabelFunction(label_str), distance_thres_(distance_thres) {}
+
+LabelMap EgoRightmostLaneLabelFunction::Evaluate(
+    const world::ObservedWorld& observed_world) const {
+  const auto& ego_agent = observed_world.GetEgoAgent();
+  const auto& ego_pos = observed_world.CurrentEgoPosition();
+
+  const auto right_lc = observed_world.GetRoadCorridor()
+                            ->GetLeftRightLaneCorridor(ego_pos)
+                            .second;
+  bool is_rightmost_lane;
+  if (right_lc) {
```

Comment:
seems like a copy and paste issue, would this also not be possible in a single evaluator/label function?

Reply:
you are right, there is a naming issue